File: app/src/main/java/com/example/ugcssample/utils/AppScheduledExecutorService.java
package com.example.ugcssample.utils;

import java.util.concurrent.Future;
import java.util.concurrent.ScheduledExecutorService;
import java.util.concurrent.ScheduledFuture;
import java.util.concurrent.TimeUnit;

public class AppScheduledExecutorService {


    private final String executorServiceName;
    private final ScheduledExecutorService worker;

    AppScheduledExecutorService(String executorServiceName, ScheduledExecutorService worker) {
        this.executorServiceName = executorServiceName;
        this.worker = worker;
    }

    public Future<?> submit(Runnable task) {
        return worker.submit(task);
    }

    public ScheduledFuture<?> schedule(Runnable command, long delay, TimeUnit unit) {
        return worker.schedule(command, delay, unit);
    }

    public ScheduledFuture<?> scheduleAtFixedRate(Runnable command, long initialDelay, long period, TimeUnit unit) {
        return worker.scheduleAtFixedRate(command, initialDelay, period, unit);
    }

    public ScheduledFuture<?> scheduleWithFixedDelay(Runnable command, long initialDelay, long delay, TimeUnit unit) {
        return worker.scheduleWithFixedDelay(command, initialDelay, delay, unit);
    }

    public String getExecutorServiceName() {
        return executorServiceName;
    }
}

File: app/src/main/java/com/example/ugcssample/utils/AppScheduledExecutorService.kt
package com.example.ugcssample.utils

import java.util.concurrent.Future
import java.util.concurrent.ScheduledExecutorService
import java.util.concurrent.ScheduledFuture
import java.util.concurrent.TimeUnit

class AppScheduledExecutorService internal constructor(
    private val executorServiceName: String?,
    private val worker: ScheduledExecutorService
                                                      ) {
    fun submit(task: Runnable?): Future<*>? {
        return worker.submit(task)
    }

    fun schedule(command: Runnable?, delay: Long, unit: TimeUnit?): ScheduledFuture<*>? {
        return worker.schedule(command, delay, unit)
    }

    fun scheduleAtFixedRate(
        command: Runnable?,
        initialDelay: Long,
        period: Long,
        unit: TimeUnit?
                           ): ScheduledFuture<*>? {
        return worker.scheduleAtFixedRate(command, initialDelay, period, unit)
    }

    fun scheduleWithFixedDelay(
        command: Runnable?,
        initialDelay: Long,
        delay: Long,
        unit: TimeUnit?
                              ): ScheduledFuture<*>? {
        return worker.scheduleWithFixedDelay(command, initialDelay, delay, unit)
    }

    fun getExecutorServiceName(): String? {
        return executorServiceName
    }
}